From 54b5913cd53f361ee84cb5f7412f7801e5d7b1c9 Mon Sep 17 00:00:00 2001
From: Takashi Iwai <tiwai@suse.de>
Date: Fri, 2 Feb 2024 17:19:47 +0100
Subject: [PATCH] hdajackretask: Fix build with gcc7

The old compiler doesn't allow a variable declaration inside switch
block.

Fixes: d349d20c127b ("hdajackretask: add support for pipewire stop/start")
Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 hdajackretask/apply-changes.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/hdajackretask/apply-changes.c b/hdajackretask/apply-changes.c
index 0d558f76f106..381ccf2516d0 100644
--- a/hdajackretask/apply-changes.c
+++ b/hdajackretask/apply-changes.c
@@ -213,9 +213,11 @@ cleanup:
 static gboolean restore_soundserver(struct soundserver* state, GError** err)
 {
     gboolean ok = FALSE;
+    gchar* clientconf;
+
     switch (state->type) {
         case PULSEAUDIO:
-            gchar* clientconf = get_pulseaudio_client_conf();
+            clientconf = get_pulseaudio_client_conf();
             if (state->was_killed && g_unlink(clientconf) != 0) {
                 g_set_error(err, quark(), 0, "%s", g_strerror(errno));
                 g_free(clientconf);
-- 
2.35.3

